transcom/mymove
# INT B-18826 Admin UI list requested office users

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18826)

## Summary

When a MilMove Admin logs in, we want to show the list of requested office users so they can have quick visibility of who has requested accounts. In future work, these users will request accounts via a form in the Office application.

This PR does the following:
- updates backend to allow for search functionality in UI
- adds UI page for listing requesting office users

**NOTE**
The admin app uses the frontend framework [react-admin](https://marmelab.com/react-admin/Readme.html) and the documentation for testing is not comprehensive and there is not a single unit test that I can work from. I am getting errors from `jest` that say I am not providing a resource, but I have stackoverflow'd & google'd and the general consensus is that `react-admin` is built to be "easy to hack, but difficult to test".

### How to test

1. Access MM as an admin
2. Log in
3. You should see the newly defaulted UI page of "Requested Office Users"
4. This should be blank for now since there are none in the db
5. Change the `status` value for some users in the `office_users` table to `REQUESTED`
6. Refresh the UI
7. You should see those users now pop up on this page

## Screenshots
When no users in `REQUESTED` status are in the db
![Screenshot 2024-03-14 at 8 43 59 AM](https://github.com/transcom/mymove/assets/136510600/ff05009e-d6b1-4003-8e2f-82d452da6ed7)

After changing status to `REQUESTED` for some users
![Screenshot 2024-03-14 at 8 44 44 AM](https://github.com/transcom/mymove/assets/136510600/de5b897e-7a2e-4bff-b091-3ce6d1ab1af5)
